### PR TITLE
[dynamo] use optimizers correctly in benchmarking

### PR DESCRIPTION
**Pull request by anijain2305**
We were not setting optimizers correctly

* This hid the issue that we see here - https://github.com/pytorch/torchdynamo/issues/1687
* This has also revealed that we are activating profilers for every dynamo optimized model call. This could affect speedup

cc @jansel @lezcano @fdrocha

### CODE REVIEW

File: benchmarks/dynamo/common.py [hunk 1]
```diff
@@ -942,10 +943,9 @@ def setup_amp(self):
             self.autocast = torch.cuda.amp.autocast
 
     def init_optimizer(self, device, params):
-        param_list = list(params)
-        if device == "cuda" and len(param_list) != 0:
+        if device == "cuda":
             # capturable is only supported on cuda at the moment
-            self.optimizer = torch.optim.Adam(param_list, capturable=True)
```
Comment (soumith) [inline benchmarks/dynamo/common.py:948]: can you remove `capturable=True` to simulate the real world where people aren't expected to set this?

Comment (anijain2305) [reply benchmarks/dynamo/common.py:948]: cc @mlazos Do you expect issues here? I am not aware of the full context.

Comment (mlazos) [reply benchmarks/dynamo/common.py:948]: I was planning on having Dynamo patch this, and I was going to remove this after the patch was finished. I haven't had time to find a good way to patch it that is intuitive though. If we disable this there will be graph breaks in the optimizer due to `item()` calls. See the context here: https://github.com/pytorch/torchdynamo/issues/1083

**Comment by mlazos** [comment]
What's the issue being solved here? It seems like the original code is really similar, and the linked issue didn't seem to have the relevant context. Is it the order of initialization?

**Comment by anijain2305** [comment]
> What's the issue being solved here? It seems like the original code is really similar, and the linked issue didn't seem to have the relevant context. Is it the order of initialization?

`optimizer` needs the pointers to the correct `model.parameters()`. Since, we make copy of the models for accuracy testing, the optimizer was pointing to some old parameters.

**Comment by voznesenskym** [comment]
Is there a way we can add a test for this? Esp as it is a fix.

File: benchmarks/dynamo/common.py (hunk 1 above)
Comment (anijain2305) [reply benchmarks/dynamo/common.py:948]: @soumith is it ok if we leave it as it is for now? And @mlazos can remove this later.

**Comment by anijain2305** [comment]
> Is there a way we can add a test for this? Esp as it is a fix.

Very hard. I consider benchmarking as a test itself. And currently our test is itself wrong :)

**Review by mlazos — APPROVED** [review]

**Review by yanboliang — APPROVED** [review]

**Comment by desertfire** [comment]
I am working on enabling HF model test on CI (https://github.com/pytorch/pytorch/pull/86792), and I am seeing a new graph break msg like this one. 
```
==========
Error: 0-20 18:06:28,306] torch._dynamo.convert_frame: [ERROR] WON'T CONVERT adam /opt/conda/lib/python3.10/site-packages/torch/optim/adam.py line 257 
due to: 
Traceback (most recent call last):
  File "/opt/conda/lib/python3.10/site-packages/torch/fx/graph_module.py", line 73, in _exec_with_source
    exec(compile(src, key, 'exec'), globals)
SystemError: excessive stack use: stack is 3430 deep
from user code:
   File "/opt/conda/lib/python3.10/site-packages/torch/optim/adam.py", line 300, in adam
    func(params,
Set torch._dynamo.config.verbose=True for more information
```

Although the models still run ok, there might be a compilation time issue here.

**Pull request closed without merging**